stl/inc/xstring

```diff
@@ -4369,47 +4727,72 @@ private:
+            auto& _My_data         = _Mypair._Myval2;
+            _My_data._Myres        = _BUF_SIZE; // SSO disabled in constexpr context
+            auto& _Al              = _Getal();
+            const pointer _New_ptr = _Al.allocate(_BUF_SIZE + 1); // throws
+            _My_data._Bx._Ptr      = _New_ptr;
+
+            _Elem* const _Raw_new = _Unfancy(_New_ptr);
+            _Traits::assign(_Raw_new, _BUF_SIZE, _Elem());
+        } else
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            _Mypair._Myval2._Myres = _BUF_SIZE - 1;
+            // the _Traits::assign is last so the codegen doesn't think the char write can alias this
+            _Traits::assign(_Mypair._Myval2._Bx._Buf[0], _Elem());
+        }
     }
 
-    void _Tidy_deallocate() noexcept { // initialize buffer, deallocating any storage
+    _CONSTEXPR20_DYNALLOC void _Tidy_deallocate() noexcept { // initialize buffer, deallocating any storage
         _Mypair._Myval2._Orphan_all();
-        if (_Mypair._Myval2._Large_string_engaged()) {
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) {
             const pointer _Ptr = _Mypair._Myval2._Bx._Ptr;
             auto& _Al          = _Getal();
             _Destroy_in_place(_Mypair._Myval2._Bx._Ptr);
             _Al.deallocate(_Ptr, _Mypair._Myval2._Myres + 1);
-        }
+        } else
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            if (_Mypair._Myval2._Large_string_engaged()) {
+                const pointer _Ptr = _Mypair._Myval2._Bx._Ptr;
+                auto& _Al          = _Getal();
+                _Destroy_in_place(_Mypair._Myval2._Bx._Ptr);
+                _Al.deallocate(_Ptr, _Mypair._Myval2._Myres + 1);
+            }
 
-        _Mypair._Myval2._Mysize = 0;
-        _Mypair._Myval2._Myres  = _BUF_SIZE - 1;
-        // the _Traits::assign is last so the codegen doesn't think the char write can alias this
-        _Traits::assign(_Mypair._Myval2._Bx._Buf[0], _Elem());
+            _Mypair._Myval2._Mysize = 0;
+            _Mypair._Myval2._Myres  = _BUF_SIZE - 1;
+            // the _Traits::assign is last so the codegen doesn't think the char write can alias this
+            _Traits::assign(_Mypair._Myval2._Bx._Buf[0], _Elem());
+        }
     }
 
 public:
-    void _Orphan_all() noexcept { // used by filesystem::path
+    _CONSTEXPR20_DYNALLOC void _Orphan_all() noexcept { // used by filesystem::path
         _Mypair._Myval2._Orphan_all();
     }
 
 private:
-    void _Swap_proxy_and_iterators(basic_string& _Right) {
+    _CONSTEXPR20_DYNALLOC void _Swap_proxy_and_iterators(basic_string& _Right) {
         _Mypair._Myval2._Swap_proxy_and_iterators(_Right._Mypair._Myval2);
     }
 
-    _Alty& _Getal() noexcept {
+    _CONSTEXPR20_DYNALLOC _Alty& _Getal() noexcept {
         return _Mypair._Get_first();
     }
 
-    const _Alty& _Getal() const noexcept {
+    _CONSTEXPR20_DYNALLOC const _Alty& _Getal() const noexcept {

```

Comment:
This line here lets the `P0896R4_views_transform_death\test.cpp` fail.

How you might ask? I have absolutely no idea

Reply:
I believe this is "fixed" by the workaround for the default constructor